src/discussions/data/hooks.js

```diff
@@ -174,3 +179,17 @@ export const useCurrentDiscussionTopic = () => {
   }
   return null;
 };
+
+export const useUserCanAddThreadInBlackoutDate = () => {
+  const blackoutDateRange = useSelector(selectBlackoutDate);
+  const isUserAdmin = useSelector(selectUserIsStaff);
+  const userHasModerationPrivilages = useSelector(selectUserHasModerationPrivileges);
+  const isUserGroupTA = useSelector(selectUserIsGroupTa);
+  const isCourseAdmin = useSelector(selectIsCourseAdmin);
+  const isCourseStaff = useSelector(selectIsCourseStaff);
+  const ifInBlackoutDateRange = inBlackoutDateRange(blackoutDateRange);
```

Comment:
Rema the const `ifInBlackoutDateRange` to `IsInBlackoutDateRange`

Reply:
date T, Role T => Add T
date F, Role T => Add T
date T, Role F => Add F
date F, Role F => Add T

!T || T => T
!F || T => T 
!T || F => F 
!F || F => T


!(InBlackoutDateRange) ||
(isUserAdmin || userHasModerationPrivilages || isUserGroupTA || isCourseAdmin || isCourseStaff)